brian2/equations/equations.py

```diff
@@ -433,6 +437,25 @@ def __init__(self, type, varname, dimensions, var_type=FLOAT, expr=None,
                                                       if variable =='xi' or variable.startswith('xi_')]),
                                     doc='Stochastic variables in the RHS of this equation')
 
+
+    _state_tuple = property(lambda self: (self.type, self.varname,
+                                          self.dim, self.var_type,
+                                          self.expr, tuple(self.flags)),
+                            doc='A tuple representing the full state of this '
+                                'object, used for hashing and equality '
+                                'testing.')
```

Comment:
Same comment again. Here we could automate the creation of ``_state_tuple`` from the arguments passed to ``__init__``, so either put a decorator there or have it derive from some class that automates the creation of the ``_state_tuple``. This solution could probably also be used for ``Variable`` actually.